Content.Shared/Silicons/Laws/SharedSiliconLawSystem.cs

```diff
@@ -21,6 +21,10 @@ public override void Initialize()
 
     protected virtual void OnAttemptEmag(EntityUid uid, EmagSiliconLawComponent component, ref OnAttemptEmagEvent args)
     {
+        //prevent self emagging
+        if (uid == args.UserUid)
+            args.Handled = true;
+
```

Comment:
You should return rather than continuing the rest of the statement, and it should also have a popup.